sdks/python/apache_beam/runners/interactive/interactive_runner.py

```diff
@@ -65,8 +65,12 @@ def __init__(self,
     """
     self._underlying_runner = (underlying_runner
                                or direct_runner.DirectRunner())
-    self._cache_manager = cache.FileBasedCacheManager(cache_dir, cache_format)
-    self._renderer = pipeline_graph_renderer.get_renderer(render_option)
+    if not ie.current_env().cache_manager():
+      ie.current_env().set_cache_manager(
+          cache.FileBasedCacheManager(cache_dir,
+                                      cache_format))
+    self._cache_manager = ie.current_env().cache_manager()
```

Comment:
Why make the CacheManager a global instead of per InteractiveRunner? The way that it is written here is that every InteractiveRunner will receive the same CacheManager. What if the user wants to create a new InteractiveRunner?

Reply:
Yes, they will share the same cache manager since the PCollections, PTransforms and Pipelines the user has defined are global.

The notebook users own their source code in notebooks, so they should always be able to take advantage of Interactive Beam features if they are using the same Pipeline and same PCollection.

Cache belongs to the implicit implementation details. So the user should not know about it or depend on it directly throughout notebook usage. The most they can do are `watch` and `show`.